### PR TITLE
migrate APIM714GetAllDocumentationTestCase

### PR DESCRIPTION
**Pull request by Krishanx92**

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM714GetAllDocumentationTestCase.java [hunk 1]
```diff
@@ -98,161 +73,83 @@ public void setEnvironment() throws Exception {
...
-        String docLocation = "";
-
-        //add document with document type="how-to" and source type=inline;
-        HttpResponse addDocumentResponse = apiPublisher.addDocument(apiName, apiVersion, apiProvider, docName, docType,
-                sourceTypeInLine, docUrl, summary, docLocation);
-        assertEquals(addDocumentResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
-                "Error in Add Document Response Code");
-        JSONObject addDocumentJsonObject = new JSONObject(addDocumentResponse.getData());
-        assertFalse(addDocumentJsonObject.getBoolean("error"), "Response Error in Add Document: " + docName);
+        restAPIPublisher
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM714GetAllDocumentationTestCase.java]: Lets use the common method to create and publish API

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM714GetAllDocumentationTestCase.java [hunk 2]
```diff
@@ -98,161 +73,83 @@ public void setEnvironment() throws Exception {
...
+                .changeAPILifeCycleStatus(createResponse.getData(), APILifeCycleAction.PUBLISH.getAction(),
+                        null);
 
-        //add document with document type="how-to" and source type=url
-        sourceDocUrl = "https://docs.wso2.com/display/AM191/Tutorials";
-        sourceTypeUrl = "url";
-        docNameWithUrl = "TestDocumentWithUrl";
+        HttpResponse applicationResponse = restAPIStore.createApplication(applicationName,
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM714GetAllDocumentationTestCase.java]: We won't be needing an application and subscription for this test

File: modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml [hunk 3]
```diff
@@ -53,6 +53,7 @@
             <class name="org.wso2.am.integration.tests.other.APIM678ApplicationCreationTestCase" />
             <class name="org.wso2.am.integration.tests.other.APIM710AllSubscriptionsByApplicationTestCase"/>
             <class name="org.wso2.am.integration.tests.other.APIMANAGER3226APINameWithDifferentCaseTestCase"/>
+            <class name="org.wso2.am.integration.tests.other.APIM714GetAllDocumentationTestCase"/>
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml]: Lets include the testng at once at the end. We can remove this for now.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM714GetAllDocumentationTestCase.java (hunk 1 above)
Comment (Krishanx92) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM714GetAllDocumentationTestCase.java]: Fixed.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM714GetAllDocumentationTestCase.java (hunk 2 above)
Comment (Krishanx92) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM714GetAllDocumentationTestCase.java]: Fixed

File: modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml (hunk 3 above)
Comment (Krishanx92) [reply modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml]: Fixed

**Pull request merged**